micrometer-support/src/main/java/io/javaoperatorsdk/operator/monitoring/micrometer/MicrometerMetrics.java

```diff
@@ -116,6 +124,14 @@ public void receivedEvent(Event event, Map<String, Object> metadata) {
   @Override
   public void cleanupDoneFor(ResourceID resourceID, Map<String, Object> metadata) {
     incrementCounter(resourceID, "events.delete", metadata);
+
+    // schedule deletion of meters associated with ResourceID
+    metersCleaner.schedule(() -> {
```

Comment:
Since this is a non trivial logic, we should add unit test.

Reply:
I'm not sure a unit test makes sense here, though I certainly could add one. What would make more sense would be an integration test to check that the metrics are actually not there anymore, will see how complex that could be to set up.

Reply:
IMO we should tests every non trivial logic with unit tests, and see at least that the API there is called for every ID eventually.

Reply:
The logic is trivial: on delete, call remove on all the meters associated with the resource id. I've added an integration test, anyway.